llm-recs/appengine-server/src/text-templates/llm_gaxois_vertexapi_palm2.ts

```diff
@@ -117,64 +121,65 @@ export async function sendPalm2Request(
   );
 }
 
-interface Palm2ApiOptions {
+export interface Palm2ApiOptions {
   modelId: string,
   apiEndpoint: string,
   requestParameters: Palm2ApiParams
 }
 
+// TODO: consider using lodash merge...
+function overideOptions(
+  oldOptions: Palm2ApiOptions,
+  newOptions?: Partial<Palm2ApiOptions>,
+): Palm2ApiOptions {
+  const finalOptions: Palm2ApiOptions = (newOptions &&
+    Object.assign({ ...oldOptions }, newOptions)) || oldOptions;
+  if (newOptions && newOptions.requestParameters) {
+    finalOptions.requestParameters =
+      Object.assign({ ...oldOptions.requestParameters }, newOptions.requestParameters);
+  }
+  return finalOptions;
+}
+
 export class VertexPalm2LLM implements LLM<Palm2ApiOptions> {
   public name: string;
-  public options: Palm2ApiOptions;
+  public defaultOptions: Palm2ApiOptions;
 
   constructor(
     public projectId: string,
     public client: AuthClient,
-    public params?: Palm2ApiOptions,
+    public initOptions?: Partial<Palm2ApiOptions>,
   ) {
-    const defaultOptions = {
-      modelId: 'text-bison',
-      apiEndpoint: 'us-central1-aiplatform.googleapis.com',
-      requestParameters: {
-        temperature: 0.7,
-        topK: 40,
-        topP: 0.95,
-        candidateCount: 4,
-        maxOutputTokens: 256,
-        stopSequences: [],
-      }
-    } as Palm2ApiOptions;
-    this.options = params ? params : defaultOptions;
-    this.name = `VertexLLM:` + this.options.modelId;
+    this.defaultOptions = overideOptions(DEFAULT_OPTIONS, initOptions);
```

Comment:
Why do we still name this as `defaultOptions` after overriding the default ones?

Reply:
they are the one's default to the instance, which can get overwritten by the predict call... ?

Reply:
re-reading, I think naming this options is good; it's the class options, and default can refer to some global default.